### ISSUE TITLE
Kubelet is upgraded before API server which breaks the upgrade

### ISSUE THREAD

**Issue by alexandre-allard** [scality/metalk8s#2845]
**Component**: salt, upgrade

**What happened**:
Upgrading a cluster from 2.5.1 to 2.6.0-dev, the process fails (most of the time), with some errors about pillar keys that do not exist (see #2832).

**What was expected**:
The upgrade works. :)

**Steps to reproduce**:
Install a MetalK8s 2.5.1, then try to upgrade to 2.6.0-dev with the following:
```
/srv/scality/metalk8s-2.6.0-dev/upgrade.sh --destination-version 2.6.0-dev
```

**Resolution proposal** (optional):
After some investigations, it looks that this is due to the fact we upgrade Kubelet on the bootstrap node prior to upgrade all the API server instances, which is not supported according to the [K8s version skew policy](https://v1-17.docs.kubernetes.io/docs/setup/release/version-skew-policy/).
Kubelet version can't be superior to the API server one and the API server version is determined by the member with the lowest version.
So, it breaks kubelet which tries to access to non-existing API resources. 
There is a quick workaround to avoid the issue by disabling the culprit feature (but we probably will encounter this issue again with later upgrades), adding the following in the kubelet configuration `/var/lib/kubelet/config.yaml`:
```
featureGates:
  CSIMigration: false
```

We have few solutions to really resolve this issue:

1. we add a boolean in the pillar (e.g. `skip_kubelet`), skipping the Kubelet upgrade during the bootstrap upgrade.
2. we remove every `metalk8s.kubernetes.kubelet` includes from the bootstrap upgrade orchestration, which means we need to add them back explicitly everywhere it is needed.
3. we change how the orchestrate works, to upgrade all API servers prior to upgrade anything else (except repos, registry and salt related stuff).

**Comment by NicolasT** [comment]
- Disabling the `CSIMigration` feature is a big no-no to me. Especially given that, over time, this will no longer be gated and then we could be stuck again (or a similar scenario could arise for non-gated features).
- I believe option 3, first upgrading all API-server instances explicitly, then continuing the whole process, is likely the 'best' option since it is explicit, and follows the upstream requirements to the letter.

**Comment by alexandre-allard** [comment]
We could also always ship Kubelet one version below APIServer, so it solves the issue for both upgrade and downgrade since up to 2 minor versions below the current one are supported.

**Comment by alexandre-allard** [comment]
Fixed by #2846
